### PR TITLE
room blacklist / custom room route cost

### PR DESCRIPTION
**Pull request by karlthepagan**
Super simple route blacklisting. This impacts many parts of the engine ;)

bestSpawnRoomFor and creep pathfinding should respect it.

Do you have a friendly room or highway that you sometimes don't have vision to? `{ 'W0N0': 1 }`

Do you have creeps that keep trying to cross the center of a quad and get slaughtered? `{ 'W4N4': 31 }`

### CODE REVIEW

**Comment by cyberblast** [comment]
uhm yes, it's fine. but we are not using room.findRoute, it's deactivated (you may activate it by setting ROUTE_PRECALCULATION to true) because it's not working as intended. Throws a lot of 'no path' errors...

**Comment by karlthepagan** [comment]
Ok to be sure, this affects every call to routeRange:
* creep.setup.maxCount for: melee, ranger, and many combat setups.
* default range finding for FlagDir
* bestSpawnRoomFor
* guard predictedRenewal

Also I now see that these are permanently cached in Memory.routeRange so that might need to get cleared whenever `ROUTE_ROOM_COST` is updated.

I'll summarize this and add to parameter.js

**Comment by karlthepagan** [comment]
Documented!

File: parameter.js [hunk 1]
```diff
@@ -78,14 +78,15 @@ var mod = {
...
     COST_MATRIX_VALIDITY: 1000,
     CONTROLLER_SIGN: false,
     CONTROLLER_SIGN_MESSAGE: "Territory of the Open Collaboration Society! (https://github.com/ScreepsOCS)",
     PLAYER_WHITELIST: ['cyberblast','SirLovi','Asku','Kazume','Noxeth','MrDave','Telemac','Xephael','Zoiah','fsck-u','FaceWound','forkmantis','Migaaresno','xAix1999','silentpoots','arguinyano','OokieCookie','OverlordQ','Nibinhilion','Crowsbane','Yew','BogdanBiv','s1akr','Pandabear41','Logmadr','Patrik','novice','Conquest','ofirl','GeorgeBerkeley','TTR','tynstar','K-C','Hoekynl'],
     // Don't attack. Must be a member of OCS for permanent whitelisting in git repository. But you can change your own copy... Please ask if you are interested in joining OCS :)
-    DEFENSE_BLACKLIST: [] // Don't defend those rooms (add room names). Blocks spawning via defense task (will not prevent offensive actions at all)
+    DEFENSE_BLACKLIST: [], // Don't defend those rooms (add room names). Blocks spawning via defense task (will not prevent offensive actions at all)
+    ROUTE_ROOM_COST: {} // custom room routing cost: e.g. `{ 'W4N4': 11 }`. Affects bestSpawnRoomFor, Creep.Setup calculations, and travel cost predictions.
```
Comment (cyberblast) [inline parameter.js]: Please add a note for the memory issue, like: 
`ROUTE_ROOM_COST: {} // custom room routing cost: e.g. `{ 'W4N4': 11 }`. Affects bestSpawnRoomFor, Creep.Setup calculations, and travel cost predictions. Please call 'delete Memory.routeRange' whenever you change this property.`

Comment (cyberblast) [inline parameter.js]: Please add a note for the memory issue, like: 
`ROUTE_ROOM_COST: {} // custom room routing cost: e.g. `{ 'W4N4': 11 }`. Affects bestSpawnRoomFor, Creep.Setup calculations, and travel cost predictions. Please call 'delete Memory.routeRange;' whenever you change this property.`

Comment (karlthepagan) [inline parameter.js]: Done.

**Pull request merged**